sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/CachedBatchColumnVector.java

```diff
@@ -0,0 +1,259 @@
+    }
+  }
+
+  //
+  // APIs dealing with nulls
+  //
+
+  @Override
+  public boolean isNullAt(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.isNullAt(ORDINAL);
+  }
+
+  //
+  // APIs dealing with Booleans
+  //
+
+  @Override
+  public boolean getBoolean(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getBoolean(ORDINAL);
+  }
+
+  @Override
+  public boolean[] getBooleans(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+
+  //
+  // APIs dealing with Bytes
+  //
+
+  @Override
+  public byte getByte(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getByte(ORDINAL);
+  }
+
+  @Override
+  public byte[] getBytes(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with Shorts
+  //
+
+  @Override
+  public short getShort(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getShort(ORDINAL);
+  }
+
+  @Override
+  public short[] getShorts(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with Ints
+  //
+
+  @Override
+  public int getInt(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getInt(ORDINAL);
+  }
+
+  @Override
+  public int[] getInts(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  public int getDictId(int rowId) {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with Longs
+  //
+
+  @Override
+  public long getLong(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getLong(ORDINAL);
+  }
+
+  @Override
+  public long[] getLongs(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with floats
+  //
+
+  @Override
+  public float getFloat(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getFloat(ORDINAL);
+  }
+
+  @Override
+  public float[] getFloats(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with doubles
+  //
+
+  @Override
+  public double getDouble(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getDouble(ORDINAL);
+  }
+
+  @Override
+  public double[] getDoubles(int rowId, int count)  {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with Arrays
+  //
+
+  @Override
+  public int getArrayLength(int rowId) {
+    throw new UnsupportedOperationException();
+  }
+  @Override
+  public int getArrayOffset(int rowId) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void loadBytes(ColumnVector.Array array) {
+    throw new UnsupportedOperationException();
+  }
+
+  //
+  // APIs dealing with Byte Arrays
+  //
+
+  public final UTF8String getUTF8String(int rowId) {
+    prepareAccess(rowId);
+    return unsafeRow.getUTF8String(ORDINAL);
+  }
+
+  private void initialize(byte[] buffer, DataType type) {
+    if (columnAccessor == null) {
```

Comment:
Is this check needed?
I guess this is called only from constructor. ditto for `mutableRow` below.